pages/blog/typescript-types.mdx

```diff
@@ -0,0 +1,145 @@
+import withArticle from '../../ui/blog/article'
+export const meta = {
+  title: 'Getting the best of TypeScript and GraphQL: union typese',
+  author: 'tuvalSimha',
+  tags: ['graphql', 'typescript', 'codegen'],
+  date: '2022-07-20',
+  description: '',
+  image: '/blog-assets/graphql-code-generator/cover.png'
+}
+export default withArticle(meta)
+ 
+# Introduction
+
+The combination between TypeScript and GraphQL is great. Both TypeScript and GraphQL support the concept of discriminated unions.
+In this short blog post, I want to share my last project experience and share one of the most powerful methods to create stronger types using TypeScript and GraphQL.
+
+One of my assumptions in this blog post is that you know some TypeScript. If you don't, you can check out ["Basic Types"](https://www.typescriptlang.org/docs/handbook/basic-types.html) handbook.
+Surprisingly, if you have some experience in TypeScript or not, you can understand the Types concept very fast. So, let's talk a little bit about TypeScript union types, one of the most remarkable features of this language.
```

Comment:
I'm not sure I understand the initial part of the sentence.
Are you surprised that whether you have the experience or not, you can understand the Types concept fast?